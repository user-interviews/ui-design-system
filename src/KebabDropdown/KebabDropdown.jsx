import React from 'react';
import PropTypes from 'prop-types';

import EllipsisIcon from '../EllipsisIcon';

const KebabDropdown = ({
  ariaLabel,
  children,
}) => (
  <span className="dropdown ui-dropdown">
    <button
      aria-expanded="false"
      aria-haspopup="true"
      aria-label={`Open ${ariaLabel}`}
      className="btn btn-link btn-link"
      data-toggle="dropdown"
      type="button"
    >
      <EllipsisIcon />
    </button>
    <span
      aria-label={ariaLabel}
      className="dropdown-menu dropdown-menu-right"
    >
      {children}
    </span>
  </span>
);

KebabDropdown.propTypes = {
  ariaLabel: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
};

export default KebabDropdown;
