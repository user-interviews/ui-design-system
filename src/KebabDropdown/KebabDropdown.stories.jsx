import React from 'react';

import 'jquery';
import 'popper.js';
import 'bootstrap/dist/css/bootstrap.min.css';
import 'bootstrap/js/dist/dropdown';

import KebabDropdown from 'src/KebabDropdown';

export default {
  title: 'Design System/KebabDropdown',
  component: KebabDropdown,
};

export const Default = () => (
  <KebabDropdown ariaLabel="Test Options">
    <div>Hello World!</div>
  </KebabDropdown>
);
