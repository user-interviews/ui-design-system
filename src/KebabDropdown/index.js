import KebabDropdown from './KebabDropdown';

export default KebabDropdown;
