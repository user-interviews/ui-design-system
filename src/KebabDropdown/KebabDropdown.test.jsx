import React from 'react';

import { render } from '@testing-library/react';

import KebabDropdown from './KebabDropdown';

describe('KebabDropdown', () => {
  const ariaLabel = 'Test options';
  const dropdownContent = 'Hello World';

  it('renders the expected snapshot', () => {
    const { container } = render(
      <KebabDropdown {...{ ariaLabel }}>
        <div>{dropdownContent}</div>
      </KebabDropdown>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  xit('shows and hides the dropdown on click', () => {
    // TODO: Test show/hide which does not work without supporting css/js
    // const dropDownButton = screen.queryByLabelIfText(ariaLabel);
  });
});
