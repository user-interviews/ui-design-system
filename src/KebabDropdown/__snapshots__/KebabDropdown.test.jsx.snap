// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KebabDropdown renders the expected snapshot 1`] = `
<span
  class="dropdown ui-dropdown"
>
  <button
    aria-expanded="false"
    aria-haspopup="true"
    aria-label="Open Test options"
    class="btn btn-link btn-link"
    data-toggle="dropdown"
    type="button"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-ellipsis-v fa-w-6 "
      data-icon="ellipsis-v"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 192 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M96 184c39.8 0 72 32.2 72 72s-32.2 72-72 72-72-32.2-72-72 32.2-72 72-72zM24 80c0 39.8 32.2 72 72 72s72-32.2 72-72S135.8 8 96 8 24 40.2 24 80zm0 352c0 39.8 32.2 72 72 72s72-32.2 72-72-32.2-72-72-72-72 32.2-72 72z"
        fill="currentColor"
      />
    </svg>
  </button>
  <span
    aria-label="Test options"
    class="dropdown-menu dropdown-menu-right"
  >
    <div>
      Hello World
    </div>
  </span>
</span>
`;
